tests/foreman/cli/test_contentcredentials.py

```diff
@@ -102,6 +102,46 @@ def test_positive_get_info_by_name(module_org):
     assert gpg_key['name'] == name
 
 
+@pytest.mark.tier1
+def test_positive_block_delete_key_in_use(module_org):
+    """Create a product and single associated repository. Create a new gpg key and associate
+        it with the product and repository. Attempt to delete the gpg key in use
+
+    :id: 022555fd-e6f2-4c95-80d7-cae26993ca8f
+
+    :expectedresults: deletion of the gpg key is blocked by raising CLIReturnCodeError 70,
+        the key is not deleted, and remains associated with the product and repository
+
+    """
+    name = gen_string('utf8')
+    product = make_product({'organization-id': module_org.id})
+    repo = make_repository({'product-id': product['id']})
+    gpg_key = make_content_credential(
+        {'key': VALID_GPG_KEY_FILE_PATH, 'name': name, 'organization-id': module_org.id}
+    )
+    # Associate repo with the product, gpg key with product and repo
+    Product.update(
+        {'gpg-key-id': gpg_key['id'], 'id': product['id'], 'organization-id': module_org.id}
+    )
```

Comment:
Anywhere that we use `Product`, `ContentCredential`, or `Repository` should be accessed through `target_sat`. 
```suggestion
    target_sat.cli.Product.update(
        {'gpg-key-id': gpg_key['id'], 'id': product['id'], 'organization-id': module_org.id}
    )
```

At the top of this test, you should pass in `target_sat` so that we have access to it's CLI class (which gives us all the hammer commands we need like `Product.update()`). If you're wondering why other tests don't do this, it's because they haven't been updated since we started adding the cli object to `target_sat`.